### PR TITLE
Comprehensive Testing

### LINKED ISSUE #308: [P1] Comprehensive Testing
- Review all the current test cases
- Add more edge cases wherever required

### PR DESCRIPTION
**Pull request by kunalsz**
Resolves #308 

## Changes Made:
- Setup of tests made easier by directly importing config from `alerting.json`. User can just update the `alerting.json` to run the tests.
- New tests added to cover edge cases,such as:
  - `test_no_alerts` : Test that no alerts are sent when there are no alerts to notify
  - `test_improper_config` :  Test that an error is raised if the configuration is not correct
  - `test_alert_network_failure` : Test that alert is not marked as notified if there are any Network Fails
- `email_alerting.py` updated to handle missing configs

### CODE REVIEW

File: buffalogs/impossible_travel/alerting/email_alerting.py [hunk 1]
```diff
@@ -40,18 +44,13 @@ def notify_alerts(self):
...
+            try:
                 send_mail(subject, body, self.email_config.get("DEFAULT_FROM_EMAIL"), self.recipient_list)  # 1 if sent,0 if not
                 self.logger.info(f"Email alert Sent: {alert.name} to {self.recipient_list}")
                 alert.notified = True
                 alert.save()
-
-        except Exception as e:
-            self.logger.error(f"Email alert failed for {alert.name}: {str(e)}")
+            except Exception as e:
+                self.logger.error(f"Email alert failed for {alert.name}: {str(e)}")
```
Comment (ManofWax) [inline buffalogs/impossible_travel/alerting/email_alerting.py:56]: I think it's bette to use `self.logger.exception(f"Email alert failed for {alert.name}` so we get also the stack trace

File: buffalogs/impossible_travel/tests/test_alert_rocketchat.py [hunk 2]
```diff
@@ -20,6 +21,13 @@ def setUp(self):
         # Create an alert
         self.alert = Alert.objects.create(name="Imp Travel", user=self.user, notified=False, description="Impossible travel detected", login_raw_data={})
 
+    def _readConfig(self):
```
Comment (ManofWax) [inline buffalogs/impossible_travel/tests/test_alert_rocketchat.py:24]: I'm writing thi comment here but it stands for each test. Is it possible to refactor the `_readConfig` so the code is not copy-pasted in each test file? maybe by putting it in some `util.py` file somewhere?

**Review by ManofWax — CHANGES_REQUESTED** [review]

File: buffalogs/impossible_travel/tests/test_alert_rocketchat.py (hunk 2 above)
Comment (kunalsz) [reply buffalogs/impossible_travel/tests/test_alert_rocketchat.py:24]: I thought about doing the same, where do you recommend putting this function in ?
Should I create a `utils.py` in the `alerting` directory ? Like that many other repeated logic can be placed in that file for future use also.

Comment (ManofWax) [reply buffalogs/impossible_travel/tests/test_alert_rocketchat.py:24]: Could it be implemented as a static method insied `BaseAlert.py` class? If there are no problems that I'm not considering I would put it there

Comment (kunalsz) [reply buffalogs/impossible_travel/tests/test_alert_rocketchat.py:24]: Sure thing,I'll do that

**Comment by kunalsz** [comment]
@ManofWax I have made the required changes, looking forward to your review
